domains/client/domain-operator/src/fetch_domain_bootstrap_info.rs

```diff
@@ -90,26 +90,21 @@ where
     }
 
     // Check each imported consensus block to get the domain instance data
-    let (domain_instance_data, domain_created_at) = 'outer: loop {
+    #[allow(clippy::never_loop)]
+    let (domain_instance_data, domain_created_at) = loop {
```

Comment:
This loop doesn't work like the old version. The old code would try each block until it found a domain instantiation block for this domain, then return the domain instance data stored by that block. If that's the behaviour we want, we should restore the missing instantiation check.

The new loop only ever runs once, on the first block in the notification stream. If the domain is instantiated, it returns the instance data. Otherwise it returns an error, even if the domain is instantiated in a future block. If that's the behaviour we want, we can remove the loop entirely.

Reply:
not ture actually.
Loop actually would run until we domain is instatiated. if the block imported return None, it means stream is closed and we return the error stopping an essential task

The loop is still required irrespective

Reply:
I see the issue. Should be fixed